src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -64,10 +69,21 @@ public SqlTriggerListener(string connectionString, string tableName, string user
             this._executor = executor ?? throw new ArgumentNullException(nameof(executor));
             this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
             this._configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
-
+            int? configuredMaxChangesPerWorker;
             // Do not convert the scale-monitor ID to lower-case string since SQL table names can be case-sensitive
             // depending on the collation of the current database.
             this._scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{userFunctionId}-SqlTrigger-{tableName}");
+
+            //In case converting from string to int is not possible from the user input.
+            try
+            {
+                configuredMaxChangesPerWorker = configuration.GetValue<int?>(SqlTriggerConstants.ConfigKey_SqlTrigger_MaxChangesPerWorker);
```

Comment:
Can you use `configuration.GetValue<int>` instead? If the configuration is absent, you should get default of int i.e., `0` as result. That will simplify the check in line 86. You will still require try-catch block for non-integer configuration values.